### ISSUE TITLE
how can I boost the performance of NGCF

### ISSUE THREAD

**Issue by SuperHenry2333** [xiangwang1223/neural_graph_collaborative_filtering#26]
Hello, I run your code on ml10m,  split into training set and test set by 7:3. Here is some of my result:
recall@5: 0.13939; precision@5: 0.23322; hit@5: 0.61446; ndcg@5: 0.25145
here are the hyperparameters I used:
regs=[1e-5], embed_size=64, layer_size=[64,64,64], lr=0.001, batch_size=1024, node_dropout=0.1, mess_dropout=[0.1,0.1,0.1]
while the result of BPR and eALS are:
BPR:
recall@5: 0.1557; precision@5: 0.2638; hit@5: 0.6589; ndcg@5: 0.2840
eALS:
recall@5:0.1695 ; precision@5: 0.2666; hit@5: 0.6735; ndcg@5: 0.2853

I wonder how I can boost the performance of NGCF. Which hyperparameters should I change and how can I change them?  I can give you any details about my experiment.

**Comment by Jancy1072** [comment]
I run the code on ml1m, and the result is also bad. I have tried some parameters but they are of no use.

**Comment by songiu** [comment]
> 您好，我在ml10m上运行您的代码，并在7：3之前分为训练集和测试集。这是我的一些结果
> ：callback @ 5：0.13939；精度@ 5：0.23322; hit @ 5：0.61446; ndcg @ 5：0.25145
> 这是我使用的超参数：
> regs = [1e-5]，embed_size = 64，layer_size = [64,64,64]，lr = 0.001，batch_size = 1024，node_dropout = 0.1，mess_dropout = [0.1 ，0.1,0.1]，
> 而BPR和eALS的结果为：
> BPR
> ：callback @ 5：0.1557；精度@ 5：0.2638; hit @ 5：0.6589; ndcg @ 5：0.2840
> eALS：
> 召回@ 5：0.1695；精度@ 5：0.2666; 5命中：0.6735; ndcg @ 5：0.2853
> 
> 我不知道如何提高NGCF的性能。我应该更改哪些超参数，以及如何更改它们？我可以为您提供有关我的实验的任何详细信息。

请问您找到解决办法了吗？

**Comment by songiu** [comment]
> 我在ml1m上运行代码，结果也很糟糕。我尝试了一些参数，但是它们没有用。

请问您找到解决的办法了吗？

**Comment by xiangwang1223** [comment]
Hi all, thanks for your interest. The possible reason is that MovieLens-1m (density is around 0.04) is much denser than the datasets reported in the paper (density is less than 0.001). When the user-item interactions are denser, involving higher-order things would influence the representation learning, and cause some popularity bias (based on our recent experiments).

There are some potential solutions:
(1) pre-train BPRMF to obtain the obtained ID embeddings, and then fine-tune the ID embeddings via NGCF.
(2) try our latest work, [LightGCN](https://github.com/kuandeng/LightGCN), which is founded on NGCF but much simpler and more effective.

Hope this is helpful.
